horovod/common/controller.cc

```diff
@@ -981,14 +985,7 @@ bool Controller::IncrementTensorCount(const Request& msg, int joined_size) {
     timeline_.NegotiateStart(name, msg.request_type());
   } else {
     std::vector<Request>& messages = table_iter->second;
-    if(msg.request_type() == Request::BARRIER) {
-      if(tensor_queue_.IsTensorPresentInTable(name)) {
-        messages.push_back(msg);
-      }
-    }
-    else {
-      messages.push_back(msg);
-    }
+    messages.push_back(msg);
```

Comment:
What was the motivation for the special handling for `BARRIER` here?

Reply:
Yea this was residue of the logic that pushed response back to the queue which you removed in [this](https://github.com/horovod/horovod/pull/3300). When we don't push it back to the queue for next cycle, this code block becomes unnecessary.

Reply:
That makes sense, thanks